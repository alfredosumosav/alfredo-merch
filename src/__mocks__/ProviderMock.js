import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import AppContext from '../context/AppContext';
import useInitialState from '../hooks/useInitialState';

const ProviderMock = ({ children }) => {
  const initialState = useInitialState();

  return (
    <AppContext.Provider value={initialState}>
      <BrowserRouter>{children}</BrowserRouter>
    </AppContext.Provider>
  );
};

export default ProviderMock;
