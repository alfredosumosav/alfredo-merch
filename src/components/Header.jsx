import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/components/Header.css';

const Header = () => (
  <div className="Header">
    <h1 className="Header-title">
      <Link to="/">Alfredo Merch</Link>
    </h1>
    <div className="Header-checkout">
      <Link to="/checkout">
        <i className="fas fa-shopping-basket" />
      </Link>
    </div>
  </div>
);

export default Header;
