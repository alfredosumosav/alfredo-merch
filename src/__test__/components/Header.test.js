/* eslint-disable no-undef */
import React from 'react';
import { mount, shallow } from 'enzyme';
import ProviderMock from '../../__mocks__/ProviderMock';
import Header from '../../components/Header';

describe('<Header />', () => {
  test('Header component renders', () => {
    const header = shallow(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(header.length).toEqual(1);
  });

  test('Header title renders', () => {
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>
    );
    expect(header.find('.Header-title').text()).toEqual('Alfredo Merch');
  });
});
