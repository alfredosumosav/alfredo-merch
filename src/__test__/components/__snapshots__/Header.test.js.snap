// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Header renders correctly 1`] = `
<div
  className="Header"
>
  <h1
    className="Header-title"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Alfredo Merch
    </a>
  </h1>
  <div
    className="Header-checkout"
  >
    <a
      href="/checkout"
      onClick={[Function]}
    >
      <i
        className="fas fa-shopping-basket"
      />
    </a>
  </div>
</div>
`;
